### DISCUSSION TITLE
How to bind to a different column from the option picker?

### DISCUSSION THREAD

**Discussion by MrChadMWood** [Budibase/budibase#12221] [Help] [answered]
Right now, option picker allows key, value mappings between two columns (label, value). This is great for the purpose of a form.
For other components though, I need the value present in a different column (neither the label or value column). It would be good if we could bind to it somehow: `{{ form-name.Fields.option-picker-name.column_a }}`

Is this possible in any way?

**Comment by lburtwistle** [comment]
Hi @MrChadMWood 

You can always go back to the data provider of the selection picker to get the entire row. Here is a JavaScript example to get the specific dataprovider record.  

**Javascript example**: (could also do it by handlebars by finding the index of the selected record.)

`let rows = $("Data Provider.Rows");

// get selected item (assuming your picker is by column name id)
let selectedItem = rows.find((item) => item['id'] == $("form-name.Fields.option-picker-name"));

// if selected item is found... display another column.
if (selectedItem)
  return selectedItem['OtherColumn'];

  return;`



You can also create a new filtered data provider by adding another data provider.  In the data field of the new data provider, scroll down to select your original data provider.  Then you can add a filter within the new data provider with your form option picker value to filter the results of the original dataprovider and display it in a table or another option picker.

![image](https://github.com/Budibase/budibase/assets/21066720/137a29d5-cca7-48a6-95d2-91f9a92e39ef)

**Comment by melohagan** [comment, marked as answer]
I had run into a similar scenario myself, and raised a feature request here: https://github.com/Budibase/budibase/issues/12423
